src/providers/hana/qgshanasourceselect.cpp

```diff
@@ -0,0 +1,674 @@
+  else if ( mSearchModeComboBox->currentText() == tr( "RegExp" ) )
+    mProxyModel._setFilterRegExp( text );
+}
+
+void QgsHanaSourceSelect::mSearchColumnComboBox_currentIndexChanged( const QString &text )
+{
+  if ( text == tr( "All" ) )
+  {
+    mProxyModel.setFilterKeyColumn( -1 );
+  }
+  else if ( text == tr( "Schema" ) )
+  {
+    mProxyModel.setFilterKeyColumn( QgsHanaTableModel::DbtmSchema );
+  }
+  else if ( text == tr( "Table" ) )
+  {
+    mProxyModel.setFilterKeyColumn( QgsHanaTableModel::DbtmTable );
+  }
+  else if ( text == tr( "Comment" ) )
+  {
+    mProxyModel.setFilterKeyColumn( QgsHanaTableModel::DbtmComment );
+  }
+  else if ( text == tr( "Geometry column" ) )
+  {
+    mProxyModel.setFilterKeyColumn( QgsHanaTableModel::DbtmGeomCol );
+  }
+  else if ( text == tr( "Type" ) )
+  {
+    mProxyModel.setFilterKeyColumn( QgsHanaTableModel::DbtmGeomType );
+  }
+  else if ( text == tr( "Feature id" ) )
+  {
+    mProxyModel.setFilterKeyColumn( QgsHanaTableModel::DbtmPkCol );
+  }
+  else if ( text == tr( "SRID" ) )
+  {
+    mProxyModel.setFilterKeyColumn( QgsHanaTableModel::DbtmSrid );
+  }
+  else if ( text == tr( "Sql" ) )
+  {
+    mProxyModel.setFilterKeyColumn( QgsHanaTableModel::DbtmSql );
+  }
+}
+
+void QgsHanaSourceSelect::mSearchModeComboBox_currentIndexChanged( const QString &text )
+{
+  Q_UNUSED( text );
+  mSearchTableEdit_textChanged( mSearchTableEdit->text() );
+}
+
+void QgsHanaSourceSelect::setLayerType( const QgsHanaLayerProperty &layerProperty )
+{
+  mTableModel.addTableEntry( layerProperty );
+}
+
+QgsHanaSourceSelect::~QgsHanaSourceSelect()
+{
+  if ( mColumnTypeThread )
+  {
+    mColumnTypeThread->requestInterruption();
+    mColumnTypeThread->wait();
+    finishList();
+  }
+
+  QgsSettings settings;
+  settings.setValue( QStringLiteral( "Windows/HanaSourceSelect/geometry" ), saveGeometry() );
+  settings.setValue( QStringLiteral( "Windows/HanaSourceSelect/HoldDialogOpen" ), mHoldDialogOpen->isChecked() );
+
+  for ( int i = 0; i < mTableModel.columnCount(); i++ )
+  {
+    settings.setValue( QStringLiteral( "Windows/HanaSourceSelect/columnWidths/%1" )
+                       .arg( i ), mTablesTreeView->columnWidth( i ) );
+  }
+}
+
+void QgsHanaSourceSelect::populateConnectionList()
+{
+  cmbConnections->blockSignals( true );
+  cmbConnections->clear();
+  cmbConnections->addItems( QgsHanaSettings::getConnectionNames() );
+  cmbConnections->blockSignals( false );
+
+  setConnectionListPosition();
+
+  btnEdit->setDisabled( cmbConnections->count() == 0 );
+  btnDelete->setDisabled( cmbConnections->count() == 0 );
+  btnConnect->setDisabled( cmbConnections->count() == 0 );
+  cmbConnections->setDisabled( cmbConnections->count() == 0 );
+}
+
+// Slot for performing action when the Add button is clicked
+void QgsHanaSourceSelect::addButtonClicked()
+{
+  mSelectedTables.clear();
+
+  for ( const QModelIndex &idx : mTablesTreeView->selectionModel()->selection().indexes() )
```

Comment:
```suggestion
  const QModelIndexList indexes = mTablesTreeView->selectionModel()->selection().indexes();
  for ( const QModelIndex &idx : indexes )
```

Reply:
Done in 32b71fc50ba7c0211acb515aec311919140e3e8b.